### PR TITLE
Hide tracking in modeline if tracking-max-mode-line-entries having = 0

### PR DESCRIPTION
**Pull request by ag91**
Solves #378 
@wasamasa I never contributed to this repository before, let me know if you would like some refactoring to match your standards.

### CODE REVIEW

**Comment by ag91** [comment]
By the way, it may interest you to see the code churn (line of codes added and deleted) of last few months: most of my code is tests luckily :)
codeChurnFrom31019

File: tracking.el [hunk 1]
```diff
@@ -340,7 +340,7 @@ to be ignored."
 This returns a list suitable for `mode-line-format'.
 If `tracking-max-mode-line-entries' is a positive integer,
 only return that many entries, ending with '+n'."
-  (if (not tracking-buffers)
+  (if (or (eq tracking-max-mode-line-entries 0) (not tracking-buffers))
```
Comment (wasamasa) [inline tracking.el:343]: Please use `=` for numerical comparison.

**Comment by ag91** [comment]
I can do that. A bit of a problem is that tracking-max-mode-line-entries
can be nil and the = operator fails for that.

Do you know if there is something better than

(or (or (null tracking-max-mode-line-entries)) (= tracking-max-mode-line-entries 0)) ...)

to handle this case? I chose eq because it works for integers  and I do
not expect users to use 0.0 in there.

**Comment by wasamasa** [comment]
How about `(or (and tracking-max-mode-line-entries (= tracking-max-mode-line-entries 0)) (not tracking-buffers))`

**Comment by wasamasa** [comment]
Thanks!

**Pull request merged**